### ISSUE TITLE
Style/Guard Clause conflicts with Metrics/LineLength in certain situations

### ISSUE THREAD

**Issue by luke-hill** [rubocop-hq/rubocop#7215]
## Expected behavior

rubocop shouldn't advise me to change my code if changing it will introduce more failures, in the code example below, the changing from multi-line to single line will break the LineLength restriction

## Actual behavior

You get into a circular loop. Change from multi-line to single line, then back to multi-line

## Steps to reproduce the problem

```
# This code breaks a 120 LineLength cop
class FooKlass
  def validate
    raise StandardError, "This has a very long description which is just under the 120char rubocop LineLength cop" if 3 + 3 == 6
  end
end

# This code breaks the guard clause cop
class FooKlass
  def validate
     if 3 + 3 == 6
       raise StandardError, "This has a very long description which is just under the 120char rubocop LineLength cop"
    end
  end
end

# The guardclause cop should check for linelength because it will always increase the linelength you use (I believe it used to do this?)
```

## RuboCop version
0.71.0

**Comment by jonas054** [comment]
@luke-hill You're not the first one who misunderstands the term _guard clause_. What the cop wants is this:
```ruby
class FooKlass
  def validate
     return unless 3 + 3 == 6

     raise StandardError, "This has a very long description which is just under the 120char rubocop LineLength cop"
    end
  end
end
```

I'm going to make a change in `Style/GuardClause` that clarifies in its message what it wants you to do. A PR is in the works.

**Comment by luke-hill** [comment]
Awesome response there jonas. Thanks. Will amend accordingly.
